### PR TITLE
Add more convenient accessors to etcpal::SockAddr

### PR DESCRIPTION
**Pull request by samkearney**

### CODE REVIEW

**Comment by samkearney** [comment]
I recommend enabling "Hide whitespace changes" for this one, there were some formatting issues that got fixed automatically by clang-format
image

File: include/etcpal/cpp/inet.h [hunk 1]
```diff
@@ -506,6 +520,101 @@ constexpr uint16_t SockAddr::port() const noexcept
...
+  return etcpal_ip_is_loopback(&addr_.ip);
+}
+
+/// \brief Whether a SockAddr contains a multicast address.
+inline bool SockAddr::IsMulticast() const noexcept
+{
+  return etcpal_ip_is_multicast(&addr_.ip);
+}
+
+/// \brief Whether a SockAddr contains a multicast address.
```
Comment (ChristianReese) [inline include/etcpal/cpp/inet.h]: Should be "Whether a SockAddr contains a wildcard address."

File: tests/unit/cpp/test_inet.cpp [hunk 2]
```diff
@@ -318,273 +318,312 @@ TEST(etcpal_cpp_inet, ip_from_string_works)
...
+}
 
-    const etcpal::SockAddr sockaddr_invalid;
-    TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+TEST(etcpal_cpp_inet, sockaddr_to_ip_works)
+{
+  const auto sockaddr = etcpal::SockAddr(0x0a0b0c0d, 8888);
+  const auto ip = sockaddr.ip();
+  TEST_ASSERT_TRUE(ip.IsV4());
+  TEST_ASSERT_EQUAL_UINT32(ip.v4_data(), 0x0a0b0c0d);
```
Comment (ChristianReese) [inline tests/unit/cpp/test_inet.cpp]: This is a super minor thing, but I would put 0x0a0b0c0d into a local const int and use that in lines 509 and 512.

**Review by ChristianReese — COMMENTED** [review]
See previous comments.

**Review by ChristianReese — APPROVED** [review]

File: tests/unit/cpp/test_inet.cpp [hunk 3]
```diff
@@ -318,273 +318,314 @@ TEST(etcpal_cpp_inet, ip_from_string_works)
...
-    TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+  const etcpal::SockAddr sockaddr_invalid;
+  TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+}
+
+TEST(etcpal_cpp_inet, sockaddr_is_link_local_works)
+{
+  const auto sockaddr = etcpal::SockAddr(etcpal::IpAddr::FromString("169.254.20.40"), 8888);
+  TEST_ASSERT_TRUE(sockaddr.IsLinkLocal());
+}
```
Comment (nwagner) [inline tests/unit/cpp/test_inet.cpp:485]: Do we need to TEST_ASSERT_FALSE for some of the other types (e.g. IsMulticast)?

**Review by nwagner — APPROVED** [review]

File: tests/unit/cpp/test_inet.cpp (hunk 3 above)
Comment (samkearney) [reply tests/unit/cpp/test_inet.cpp:485]: I've added your requested tests on the develop branch in 2eb339c.

**Pull request merged**